include/alpaka/kernel/TaskKernelGenericSycl.hpp

```diff
@@ -28,8 +29,109 @@
 
 #ifdef ALPAKA_ACC_SYCL_ENABLED
 
+#if BOOST_COMP_CLANG
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wunused-lambda-capture"
+#    pragma clang diagnostic ignored "-Wunused-parameter"
+#endif
+
+#    include "alpaka/kernel/sycl_subgroup_size.h"
```

Comment:
Isn't it better to keep this inside the `#ifdef ALPAKA_ACC_SYCL_ENABLED` block?

Reply:
In principle it should work also moving this file above since there are only some macro checks and definitions. However I agree on keeping it inside the `#ifdef` because is meaningful only for SYCL (or move it above and add the `#ifdef ALPAKA_ACC_SYCL_ENABLED` in it as well)